lib/developer_portal/app/controllers/developer_portal/admin/account/personal_details_controller.rb

```diff
@@ -49,16 +45,17 @@ def deny_unless_can_update
     end
   end
 
-  def redirect_path
-    if params[:origin] == "users"
-      if current_account.provider?
-        provider_admin_account_users_path
-      else
-        admin_account_users_path
-      end
-    else
-      admin_account_personal_details_path
-    end
+  def user_params
+    params.require(:user).permit([:current_password] +
+                                   resource.special_fields +
+                                   resource.defined_fields.map(&:name))
   end
 
+  def verify_current_password
+    return unless current_user.using_password?
+    return if current_user.authenticated?(user_params[:current_password])
+
+    flash.now[:error] = 'Current password is incorrect.'
+    resource.errors.add(:current_password, 'Current password is incorrect.')
```

Comment:
```suggestion
    resource.errors.add(:base, :current_password_incorrect)
    flash.now[:error] = resource.errors.full_messages.to_sentence
```


This requires also adding this to `en.yml` on line `:935`

```
          current_password_incorrect: "Current password is incorrect."
```

(`activerecord.errors.models.user.current_password_incorrect`)

Reply:
But this can be done in another PR, this should be transparent for the user.